exchange/bidder.go

```diff
@@ -255,13 +272,13 @@ func getNativeImpByImpID(impID string, request *openrtb.BidRequest) (*openrtb.Na
 	return nil, errors.New("Could not find native imp")
 }
 
-func getAssetByID(id int64, assets []nativeRequests.Asset) nativeRequests.Asset {
+func getAssetByID(id int64, assets []nativeRequests.Asset) (*nativeRequests.Asset, error) {
 	for _, asset := range assets {
 		if id == asset.ID {
-			return asset
+			return &asset, nil
 		}
 	}
-	return nativeRequests.Asset{}
+	return nil, fmt.Errorf("Couldn't find asset with ID:%d in the request", id)
```

Comment:
Consider rewording to a slightly more professional message. Something along the lines of "Unable to find asset with ID:%d in the request."